server/controllers/login.js

```diff
@@ -1,38 +1,43 @@
 const getUserByEmail = require('../database/queries/db_user');
 const { compareHashed, loginValidation } = require('../helper');
+const { sign } = require('jsonwebtoken');
+const { SECRET } = process.env;
 
 const Login = (request, response) => {
 	let { user } = request.body;
+	const { email, password } = user;
 	const { error } = loginValidation(user);
 	if (typeof error !== 'undefined') {
 		return response.send(error.toString().replace('validationError:', ''));
 	}
-	getUserByEmail(user)
-		.then((result) => {
-			if (result.rowCount === 0)
-				return response.send('no user exist in the db');
-			let data = result.rows[0];
-			if (data !== null) {
-				compareHashed(user.password, data.password)
-					.then((result) => {
-						if (!result) {
-							response.send('no user exist in the db');
-						}
+	getUserByEmail(email, (error, result) => {
+		console.log('QUERY RESULT!', result);
+		if (result.rowCount === 0) return response.send('no user exist in the db');
+		const data = result.rows[0];
+		if (data !== null) {
+			compareHashed(password, data.password)
+				.then((result) => {
+					console.log('PASSWORD RESULT!', result);
+					if (result) {
+						const token = sign({ email }, SECRET);
 						let info = {
 							user_id: data.user_id,
 							name: data.name,
 							email: data.email,
 						};
-						response.send(info);
-					})
-					.catch((err) => {
-						throw new Error(err);
-					});
-			}
-		})
-		.catch((err) => {
-			throw new Error(err);
-		});
+						response
+							.status(200)
+							.cookie('token', token, { maxAge: 900000, httpOnly: true })
+							.json({ info, token });
+					} else {
+						response.status(400).json({ message: 'incorrect password' });
+					}
```

Comment:
Might be fun to think about how this could be changed to avoid the nesting, but might be out of scope for these changes.